```
Description: generation of training and benchmark sets for AlphaFold2 applications
Requirements: 
	- Voronota
	- PSI-BLAST
	- Foldseek 
	- gemmi
Further requirements
	- PDB fasta file
	- SwissProt fasta file	
	- SIFTS pdb chain and uniprot mapping
	- BIOGRID (in mitab format)
	- cutoff_dates.txt with cutoff dates

Note: Scripts were optimized for HPCs with slurm.
Note: Define paths in the parameters.txt file.

1. Create a list of PDBs with release date and experimental method. 
python3 script/1_pdb_summary.py

2. Create foldseek library
foldseek databases PDB pdb tmp

3. Create blast db for PDB (download from https://files.wwpdb.org/pub/pdb/derived_data/. It is recommended to use only protein sequences)
makeblastdb -in [FASTA] -dbtype prot

5. define cutoff dates (supplied, users can modify)

6. For any PDB code, search for homologous structures and sequences.
python3 script/2_batch_struct.py collects data and use slurm to distribute job 

7. Collect results. 
python3 script/3_collect_struct.py

8. Within new PDB chains search for interactions. 
python3 script/4_batch_voronota.py

9. Collect voronota results. 
python3 script/5_collect_voronota.py

10. For all SwissProt files, search for homologous sequences. 
python3 script/6_batch_seq.py collects data and use slurm to distribute job

11. Collect results. 
python3 script/7_collect_sequences.py

12. Get interactions from BIOGRID. 
python3 script/8_sequence_interactions.py


NOTES:
the code was optimized for slurm. For other environments or non-parallel execution the following modifications are required

1. Replacing step 6 
run script/structure_search.py for all pdb codes. 

Fasta file containing protein sequences for pdb_code need to be generated by the user. Cif files has to be named as pdbcode.cif, as foldseek use the filename when listing results

python3 structure_search.py [pdb_code] [pdb_code_fasta] [pdb_blast_db] [psi-blast_binary] [pdb_cif] [foldseek_pdb_lib] [foldseek_binary] [dates] [pdb_summary] [workdir] [output_file]

[pdb_code]: 4 character code
[pdb_code_fasta]: fasta file containing protein sequences for pdb_code 
[pdb_blast_db]: generated on step 3
[psi-blast_binary]: path for psi-blast
[pdb_cif]: pdb cif file (if you are running pdb codes one by one you need to generate this for yourself and name it as pdb_code.cif as foldseek use the filename when listing results!)
[foldseek_pdb_lib]: generated in step 2
[foldseek_binary]: path for foldseek
[dates]: supplied, or defined is step 5
[pdb_summary]: generated in step 1
[workdir]: working directory
[output file]: output file

2. Replacing step 8
run script/voronota.py
python3 voronota.py [pdb_code] [infile] [outfile]
[pdb_code]: 4 character code
[infile]: cif file for assemblies from PDB (containing most probable oligomerization state)
[outfile]: outfile

3. Replacing step 10
run script/sequence_search.py for all uniprot sequences

Fasta file containing protein sequences for UniProt AC need to be generated by the user.

python3 sequence_search.py [uniprot ac] [uniprot fasta] [pdb_blast_db] [psi_blast_binary] [sifts] [dates] [pdb_result] [pdb_summary] [workdir] [output]
[uniprot_ac]: UniProt AC
[uniprot_fasta]: UniProt sequence (single)
[pdb_blast_db]: generated on step 3
[psi_blast_binary]: path for psi-blast
[sifts]: SIFTS file
[dates]: supplied, or defined is step 5
[pdb_result]: generated in step 7 (STR_SEARCH_RESULTS in parameters.txt)
[pdb_summary]: generated in step 1, normally in WORKDIR/pdb_summary.txt
[workdir]: working directory
[output]: output file



Running the code for user supplied sequences:
Step 10 can be executed with any sequence files. Please note that in the UniProt fasta file the header content is separated by '|' characters, where the 2nd column contains the UniProt AC. Therefore, for costum sequences the fasta header format should be the same, or the fasta reading files has to be modified in 6_batch_seq.py,7_collect_sequences.py,8_sequence_interactions.py.
```
